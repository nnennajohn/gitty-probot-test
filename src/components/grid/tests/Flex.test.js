// @flow
import React from 'react';
import { create as render } from 'react-test-renderer';
import { Flex } from '../index';

describe('Flex Base Component', () => {
  // Flex
  test('Flex renders', () => {
    const flex = render(<Flex />).toJSON();
    expect(flex).toMatchSnapshot();
  });

  test('Flex renders with props', () => {
    const flex = render(
      <Flex
        flexWrap="wrap"
        flexDirection="column"
        alignItems="center"
        justifyContent="space-between"
      />,
    ).toJSON();
    expect(flex).toMatchSnapshot();
  });

  test('Flex renders with legacy props', () => {
    const flex = render(
      <Flex
        wrap
        flexDirection="column"
        align="center"
        justify="space-between"
      />,
    ).toJSON();
    expect(flex).toMatchSnapshot();
  });

  test('Flex renders with flexDirection prop', () => {
    const flex = render(<Flex flexDirection="column" />).toJSON();
    expect(flex).toMatchSnapshot();
  });

  test('Flex renders with responsive props', () => {
    const flex = render(
      <Flex
        wrap={[true, false]}
        flexDirection={['column', 'row']}
        align={['stretch', 'center']}
        justify={['space-between', 'center']}
      />,
    ).toJSON();
    expect(flex).toMatchSnapshot();
  });
});
