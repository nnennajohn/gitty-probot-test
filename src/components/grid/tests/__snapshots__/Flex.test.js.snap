// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Base Component Flex renders 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media only screen and (min-width:40em) {
  .c0 {
    max-width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    max-width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    max-width: 76rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Flex Base Component Flex renders with flexDirection prop 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media only screen and (min-width:40em) {
  .c0 {
    max-width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    max-width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    max-width: 76rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Flex Base Component Flex renders with legacy props 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media only screen and (min-width:40em) {
  .c0 {
    max-width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    max-width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    max-width: 76rem;
  }
}

<div
  className="c0"
  wrap={true}
/>
`;

exports[`Flex Base Component Flex renders with props 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media only screen and (min-width:40em) {
  .c0 {
    max-width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    max-width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    max-width: 76rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Flex Base Component Flex renders with responsive props 1`] = `
.c0 {
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;
}

@media only screen and (min-width:40em) {
  .c0 {
    max-width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    max-width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    max-width: 76rem;
  }
}

<div
  className="c0"
  wrap={
    Array [
      true,
      false,
    ]
  }
/>
`;
